File: DKR_1/DKR_1.py
import math
x = float(input('Введите значение х: '))
error = False
if x < -6:
    if x <= -6:
        error = True
    else:
        y = math.pow(x, 3) * math.pow(x, (0.1 * x)) - math.cos(x)
elif -6 <= x < -5:
    if x <= -6 and -5 < x:
        error = True
    else:
        y = math.pow(math.exp(x), x) - (math.sin(x) / math.cos(x))
elif -5 <= x < 5:
    if x <= -5:
        error = True
    else:
        y = math.pow(x, (0.1 * x)) / 45
else:  # x >= 5
    if x <= 5:
        error = True
    else:
        y = math.pow(x, 2) / x - math.pow(math.exp(x), x) / (math.sin(x) / math.cos(x))
if error:
    print('значение x не подходит условию')
else:
    print(f'{y:.2f}')

File: DKR_1/DKR_1.pas
uses CRT;
var x,y:real;
    error:boolean;
begin
  writeln('ведите значение х');
  readln(x);
  error:=false;
  if (x<-6) then
  begin
    if x<=-6 then
      error:=true
    else
      y:=power(x,3)*power(x,(0.1*x))-cos(x);
  end
  else if (-6<=x) and (x<-5) then
  begin
    if (x<=-6) and (-5<x) then
      error:=true
    else
       y:= power(exp(x),x)-(sin(x)/cos(x));
  end
  else if (-5<=x) and (x<5) then
  begin
    if (x<=-5) then
      error:=true
    else
       y:= power(x,(0.1*x))/45
    end
       else 
       begin
         if (x<=5) then 
           error:=true
         else
            y:= power(x,2)/x-power(exp(x),x)/(sin(x)/cos(x));
         end;
       if error then
         writeln('значение x не подходит условию')
       else writeln(y:0:2);
  end.